### PR TITLE
Setting OpenMRS ID as preferred identifier to fix error when editing …

### PR DESCRIPTION
**Pull request by njorocs**
…registration information

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@Component("kenyaemr.chore.UpdateOpenMRSIDAsPreferredIdentifier")
+public class UpdateOpenMRSIDAsPreferredIdentifier extends AbstractChore {
+
+    /**
+     * @see AbstractChore#perform(PrintWriter)
+     */
+
+    @Override
+    public void perform(PrintWriter out) {
+        String updateConceptSql = "update patient_identifier pi set pi.preferred = 1 where pi.preferred = 0 and pi.identifier_type = 3;";
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java:30]: Do you mind joining this to indentifier type table and use the uuid for openmrs ID?

Comment (ojwanganto) [reply api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java:30]: Also do this only where preferred is null

**Review by ojwanganto — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java (hunk 1 above)
Comment (njorocs) [reply api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java:30]: > Also do this only where preferred is null

The only values are 0 and 1. Edit is failing when preferred is 0

Comment (njorocs) [reply api/src/main/java/org/openmrs/module/kenyaemr/chore/UpdateOpenMRSIDAsPreferredIdentifier.java:30]: > Do you mind joining this to indentifier type table and use the uuid for openmrs ID?
okay.

**Review by patryllus — APPROVED** [review]
The default is 0 not null

**Review by patryllus — APPROVED** [review]

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**